### ISSUE TITLE
[BUG] Spring Boot application is not starting if JobRunr is used with Spring Cloud Stream RabbitMQ depenency

### ISSUE THREAD

**Issue by anowak-ct** [jobrunr/jobrunr#130]
**Describe the bug**
There is a conflict with bean "jsonMapper" if JobRunnr and Spring Cloud Stream Binder Rabbit are running in same app.

**Environment**
I'm using version 3.0.0
I'm running on JRE / JDK 11
I'm using the following StorageProvider: n/a
I'm using Spring Cloud version 2020.0.2

**To Reproduce**
Add the following dependencies in pom.xml

```
<dependency>
            <groupId>org.springframework.cloud</groupId>
            <artifactId>spring-cloud-stream-binder-rabbit</artifactId>
</dependency>
<dependency>
            <groupId>org.jobrunr</groupId>
            <artifactId>jobrunr-spring-boot-starter</artifactId>
            <version>3.0.0</version>
</dependency>
```

Error on Spring Boot startup:

```
2021-05-26 18:45:57.103  WARN 19524 --- [           main] s.c.a.AnnotationConfigApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.support.BeanDefinitionOverrideException: Invalid bean definition with name 'jsonMapper' defined in class path resource [org/springframework/cloud/function/context/config/ContextFunctionCatalogAutoConfiguration$JsonMapperConfiguration.class]: Cannot register bean definition [Root bean: class [null]; scope=; abstract=false; lazyInit=null; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=org.springframework.cloud.function.context.config.ContextFunctionCatalogAutoConfiguration$JsonMapperConfiguration; factoryMethodName=jsonMapper; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [org/springframework/cloud/function/context/config/ContextFunctionCatalogAutoConfiguration$JsonMapperConfiguration.class]] for bean 'jsonMapper': There is already [Root bean: class [null]; scope=; abstract=false; lazyInit=null; autowireMode=3; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=org.jobrunr.autoconfigure.JobRunrAutoConfiguration$JobRunrJacksonAutoConfiguration; factoryMethodName=jacksonJsonMapper; initMethodName=null; destroyMethodName=(inferred); defined in class path resource [org/jobrunr/autoconfigure/JobRunrAutoConfiguration$JobRunrJacksonAutoConfiguration.class]] bound.

2021-05-26 18:45:57.116  INFO 19524 --- [           main] ConditionEvaluationReportLoggingListener : 

Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
2021-05-26 18:45:57.132 ERROR 19524 --- [           main] o.s.b.d.LoggingFailureAnalysisReporter   : 

***************************
APPLICATION FAILED TO START
***************************

Description:

The bean 'jsonMapper', defined in class path resource [org/springframework/cloud/function/context/config/ContextFunctionCatalogAutoConfiguration$JsonMapperConfiguration.class], could not be registered. A bean with that name has already been defined in class path resource [org/jobrunr/autoconfigure/JobRunrAutoConfiguration$JobRunrJacksonAutoConfiguration.class] and overriding is disabled.

Action:

Consider renaming one of the beans or enabling overriding by setting spring.main.allow-bean-definition-overriding=true
```

This error occurs with different spring cloud and JobRunnr version combinations.

Enabling allow-bean-definition-overriding yields an error later in the startup.

**Expected behavior**
Spring application should startup without error

**Comment by rdehuyss** [comment]
Hi @anowak-ct , this should be solved in release 3.0.1. 

Enjoy!

**Comment by anowak-ct** [comment]
That's awesome. Just tested it and JobRunr & Stream/Rabbit working together with no problem!  

Thank you so much for the super fast fix. Luv using JobRunr in my project.
